File: Task-03/Sub-Task 4/Go/Ques_4.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func generateDiamond(n int, writer *bufio.Writer) {
	for i := 1; i <= n; i += 2 {
		for j := 0; j < (n-i)/2; j++ {
			writer.WriteString(" ")
		}
		for k := 0; k < i; k++ {
			writer.WriteString("*")
		}
		writer.WriteString("\n")
	}
	for i := n - 2; i > 0; i -= 2 {
		for j := 0; j < (n-i)/2; j++ {
			writer.WriteString(" ")
		}
		for k := 0; k < i; k++ {
			writer.WriteString("*")
		}
		writer.WriteString("\n")
	}
}

func main() {
	inputFile, err := os.Open("input.txt")
	if err != nil {
		fmt.Println("Error opening input.txt:", err)
		return
	}
	defer inputFile.Close()

	var n int
	scanner := bufio.NewScanner(inputFile)
	if scanner.Scan() {
		n, err = strconv.Atoi(scanner.Text())
		if err != nil {
			fmt.Println("Error converting input to integer:", err)
			return
		}
	}

	outputFile, err := os.Create("output.txt")
	if err != nil {
		fmt.Println("Error opening output.txt:", err)
		return
	}
	defer outputFile.Close()

	writer := bufio.NewWriter(outputFile)

	generateDiamond(n, writer)

	writer.Flush()

	fmt.Println("Diamond pattern written to output.txt")
}

File: Task-03/Sub-Task 4/C/Ques_4.c
#include <stdio.h>

void generateDiamond(int n, FILE *outputFile) {
    for (int i = 1; i <= n; i += 2) {
        for (int j = 0; j < (n - i) / 2; j++) {
            fprintf(outputFile, " ");
        }
        for (int k = 0; k < i; k++) {
            fprintf(outputFile, "*");
        }
        fprintf(outputFile, "\n");
    }
    for (int i = n - 2; i > 0; i -= 2) {
        for (int j = 0; j < (n - i) / 2; j++) {
            fprintf(outputFile, " ");
        }
        for (int k = 0; k < i; k++) {
            fprintf(outputFile, "*");
        }
        fprintf(outputFile, "\n");
    }
}

int main() {
    int n;
    FILE *inputFile = fopen("input.txt", "r");
    if (inputFile == NULL) {
        printf("Error opening input.txt\n");
        return 1;
    }
    fscanf(inputFile, "%d", &n);
    fclose(inputFile);

    FILE *outputFile = fopen("output.txt", "w");
    if (outputFile == NULL) {
        printf("Error opening output.txt\n");
        return 1;
    }
    generateDiamond(n, outputFile);
    fclose(outputFile);

    printf("Diamond pattern written to output.txt\n");
    return 0;
}
